File: problems/1742-maximum-number-of-balls-in-a-box/solution.py
class Solution:
    def countBalls(self, lowLimit: int, highLimit: int) -> int:
        box_counts = {}
        for i in range(lowLimit, highLimit + 1):
            s = sum(int(digit) for digit in str(i))
            if s in box_counts:
                box_counts[s] += 1
            else:
                box_counts[s] = 1
        
        max_balls = 0
        for count in box_counts.values():
            if count > max_balls:
                max_balls = count
        
        return max_balls

File: problems/1742-maximum-number-of-balls-in-a-box/solution.c
int countBalls(int lowLimit, int highLimit) {
    int counts[46] = {0};
    int maxCount = 0;
    for (int i = lowLimit; i <= highLimit; i++) {
        int sum = 0;
        int num = i;
        while (num > 0) {
            sum += num % 10;
            num /= 10;
        }
        counts[sum]++;
        if (counts[sum] > maxCount) {
            maxCount = counts[sum];
        }
    }
    return maxCount;
}